pennylane/transforms/defer_measurements.py

```diff
@@ -84,34 +92,65 @@ def qfunc(par):
 
     >>> qml.grad(qnode)(par)
     tensor(-0.49622252, requires_grad=True)
+
+    Reusing and reseting measured wires will work as expected with the
+    ``defer_measurements`` transform:
+
+    .. code-block:: python3
+
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def func(x, y):
+            qml.RY(x, wires=0)
+            qml.CNOT(wires=[0, 1])
+            m_0 = qml.measure(1, reset=True)
+
+            qml.cond(m_0, qml.RY)(y, wires=0)
+            qml.RX(np.pi/4, wires=1)
+            return qml.probs(wires=[0, 1])
+
+    Executing this QNode:
+
+    >>> pars = np.array([0.643, 0.246], requires_grad=True)
+    >>> func(*pars)
+    tensor([0.76960924, 0.13204407, 0.08394415, 0.01440254], requires_grad=True)
     """
-    measured_wires = {}
 
     cv_types = (qml.operation.CVOperation, qml.operation.CVObservable)
     ops_cv = any(isinstance(op, cv_types) for op in tape.operations)
     obs_cv = any(isinstance(getattr(op, "obs", None), cv_types) for op in tape.measurements)
     if ops_cv or obs_cv:
         raise ValueError("Continuous variable operations and observables are not supported.")
 
-    for op in tape:
-        op_wires_measured = set(wire for wire in op.wires if wire in measured_wires.values())
-        if len(op_wires_measured) > 0:
-            raise ValueError(
-                f"Cannot apply operations on {op.wires} as the following wires have been measured already: {op_wires_measured}."
-            )
+    # Current wire in which pre-measurement state will be saved if dev_wires not specified
+    cur_wire = max(tape.wires) + 1
+    new_wires = {}
 
+    for op in tape.circuit:
         if isinstance(op, MidMeasureMP):
-            measured_wires[op.id] = op.wires[0]
+            new_wires[op.id] = cur_wire
+            qml.CNOT([op.wires[0], cur_wire])
+
+            if op.reset:
+                qml.CNOT([cur_wire, op.wires[0]])
```

Comment:
Shouldn't `op.reset is True` indicate that we don't need to apply any CNOTs at all?

Reply:
I don't follow. Measurement causes measured state (let's say on wire 0) to be stored in a new wire (say 3) using `CNOT([0, 3])`. If `op.reset is True`, then we want wire 0 back in $|0\rangle$, which we would achieve by applying `CNOT([3, 0])`.

Reply:
I've opened another PR that will address this change after 0.32 is released. For now, the current implementation will remain.